### DISCUSSION TITLE
deepin-terminal增加【双击指定选择时使用的分隔符】的功能

### DISCUSSION THREAD

**Discussion by liwl1991** [linuxdeepin/developer-center#4084] [Features Request & Ideas | 特性请求 & 头脑风暴]
![image](https://user-images.githubusercontent.com/28835628/231954933-f8254687-d1e2-4576-91f1-92c5bcdf80b3.png)
类似xshell提供的这样的功能。

问题描述：当前版本的deepin-terminal，使用了第三方库默认的分隔规则，但未提供用户自定义设置接口。
这样在用户使用过程中，通过双击选择字符串时，可能会出现复制多于字符串，或者被分隔符号截断的现象，比如复制user@localhost，比如find . -name "*.cpp"|xargs egrep -n 根据关键字查询到某个文件的某行，双击复制时，会连带复制了::等问题，在使用上不够灵活
我已经实现了能够自定义分隔符的功能，类似xshell这种可给用户配置空间的特性（默认部分，可编辑设置），不清楚此特性是否符合deepin-terminal产品定位
个人初步实现的效果如下，但我这个是白名单，在编辑框的是非分隔符，不在的就是分隔符：
![image](https://user-images.githubusercontent.com/28835628/231956253-ccfcf30e-41e2-4232-abb3-989ea6c13fe9.png)

**Comment by liwl1991** [comment]
@meiyixiang @ArchieMeng

**Comment by meiyixiang** [comment]
看你的描述你是准备像我们提交PR吗？这个功能我看kde终端中是有这个功能的，这种功能主要是满足个性化需求不会对应用本身方向有什么破坏。

**Reply by liwl1991** [reply]
是的，是准备提交一个pr的。但是这个设置，目前我不清楚放在设置界面哪里比较合适，能否给一些建议

**Reply by liwl1991** [reply]
pr已经提交
